components/brave_wallet/browser/solana_response_parser.cc

```diff
@@ -274,6 +274,40 @@ bool ParseGetBlockHeight(const base::Value& json_value,
   return base::StringToUint64(*block_height_string, block_height);
 }
 
+bool ParseGetTokenAccountsByOwner(
+    const base::Value& json_value,
+    std::vector<absl::optional<SolanaAccountInfo>>* accounts) {
+  DCHECK(accounts);
+
+  auto result = ParseResultDict(json_value);
+  if (!result)
+    return false;
+
+  const auto* account_dicts_list = result->FindList("value");
+  if (!account_dicts_list)
+    return false;
+
+  // For each account dict, get the "account" key and parse the value as an
+  // SolanaAccountInfo
+  for (const auto& account_dict : *account_dicts_list) {
+    const auto* account_dict_value = account_dict.GetIfDict();
+    if (!account_dict_value)
+      return false;
+
+    const auto* account_value = account_dict_value->Find("account");
+    if (!account_value)
+      return false;
+
+    absl::optional<SolanaAccountInfo> account_info;
+    if (!ParseGetAccountInfoPayload(account_value->GetDict(), &account_info))
+      return false;
+
+    accounts->push_back(std::move(account_info));
+  }
+
+  return true;
+}
+
```

Comment:
It seems to me we should return `std::vector<SolanaAccountInfo>*` here(no optional)
`ParseGetAccountInfoPayload` also called to parse `getAccountInfo` response which may have no account found for pubkey https://docs.solana.com/developing/clients/jsonrpc-api#results
For `ParseGetTokenAccountsByOwner` null accounts are not expected and we should just skip them.